Source/CesiumEditor/Private/CesiumIonSession.cpp

```diff
@@ -127,55 +130,85 @@ void CesiumIonSession::connect() {
               true);
         }
 
-        int64_t clientID = pServer->OAuth2ApplicationID;
+        // Make request to /appData to learn the server's authentication mode
+        return thiz->ensureAppDataLoaded();
+      })
+      .thenInMainThread(
+          [ionServerUrl, thiz, pServer = this->_pServer](bool loadedAppData) {
+            if (!loadedAppData || !thiz->_appData.has_value()) {
+              Promise<Connection> promise =
+                  thiz->_asyncSystem.createPromise<Connection>();
+              promise.reject(std::runtime_error(
+                  "Failed to obtain _appData, can't create connection"));
+              return promise.getFuture();
+            }
+
+            if (thiz->_appData->needsOauthAuthentication()) {
+              int64_t clientID = pServer->OAuth2ApplicationID;
+              return CesiumIonClient::Connection::authorize(
+                  thiz->_asyncSystem,
+                  thiz->_pAssetAccessor,
+                  "Cesium for Unity",
+                  clientID,
+                  "/cesium-for-unity/oauth2/callback",
```

Comment:
Seems this was copy/pasted from Cesium for Unity. It completely breaks Cesium ion login in Cesium for Unreal because the redirect URI and client ID don't match.